### DISCUSSION TITLE
Reactive hibernate Identity provider

### DISCUSSION THREAD

**Discussion by b0c1** [quarkusio/quarkus#22727] [Q&A]
Hi guys!

I try to use quarkus with reactive hibernate (with Panache) in custom identity provider (because the security-jpa provider is not reactive and not work).
I try to do:

```java
@ApplicationScoped
public class ReactiveUsernamePasswordAuthentication implements IdentityProvider<UsernamePasswordAuthenticationRequest> {
	@Inject
	private RequestScopedUserService userService;

	@Override
	public Class<UsernamePasswordAuthenticationRequest> getRequestType() {
		return UsernamePasswordAuthenticationRequest.class;
	}

	@Override
	public Uni<SecurityIdentity> authenticate(UsernamePasswordAuthenticationRequest request,
			AuthenticationRequestContext context) {
		return userService.getUserIdentity(request);
	}
}
```

but when I try to call it I got a `javax.enterprise.context.ContextNotActiveException` because the `Arc.container().requestContext().isActive()`  is `false`
How can I propagate the request context down to the identityprovider in a reactive environment?

the user service is simple:
```java
	public Uni<SecurityIdentity> getUserIdentity(UsernamePasswordAuthenticationRequest usernamePasswordAuthenticationRequest) {
		return UserEntity
				.findByUsername(usernamePasswordAuthenticationRequest.getUsername())
				.onItem().transform(entity -> checkPassword(entity, usernamePasswordAuthenticationRequest))
				.onItem().transform(QuarkusSecurityIdentity.Builder::build)
				;
	}
```

Thanks

[a comment by an automated account is not shown]

**Comment by Sanne** [comment]
There's a current limitation of Panache requiring a request scope to define the boundaries of the underlying Hibernate resources; I know some people (including me) have some prototypes about how to improve this but it's not entirely ready and I'm not confident enough to provide a recommendation still.

I would suggest don't use Panache Reactive? Inject a `Mutiny.SessionFactory` into your service instead: this will be slightly more verbose (as you have to create the Query using "regular" non-panache code) but it would allow you to open & close sessions explicitly, which seems more natural IMO in reactive code.

**Comment by sberyozkin** [comment]
May be you can use 
```
return UserEntity
				.findByUsername(usernamePasswordAuthenticationRequest.getUsername())
				.onItem().transform(entity -> checkPassword(entity, usernamePasswordAuthenticationRequest))
				.onItem().transform(QuarkusSecurityIdentity.Builder::build)
				;
```

directly without the injection ?
